src/nvim/channel.c

```diff
@@ -517,6 +517,45 @@ uint64_t channel_from_stdio(bool rpc, CallbackReader on_output, const char **err
   return channel->id;
 }
 
+size_t newline_adjust(char **str)
+{
+#ifdef UNIX
+  return 0;
+#else
+  if (!str || !*str) {
+    return 0;
+  }
+  size_t c = 0;
+  char *p = *str;
+
+  while (*p++) {
+    c += *p == '\n' ? 1 : 0;
+  }
+  if (!c) {
+    return c;
+  }
+
+  p = *str;
+
+  char *tok = NULL;
+  char *con = NULL;
+  size_t len = STRLEN(p) + 1 + c;
+
+  char *const ret = xmalloc(len);
+  tok = os_strtok(p, "\n", &con);
+  xstrlcpy(ret, tok, len);
+  xstrlcat(ret, "\r\n", len);
+  while ((tok = os_strtok(NULL, "\n", &con))) {
+    xstrlcat(ret, tok, len);
+    xstrlcat(ret, "\r\n", len);
```

Comment:
(Leaving this for myself when I get home) If the last character in `*str` isn't `\n`, then this will add an extra two characters. The value of `c` may then be incorrect.